.gitlab-ci.yml

```diff
@@ -0,0 +1,42 @@
+stages:
+  - build
+  - trigger
+
+default:
+  interruptible: true
+
+workflow:
+  rules:
+    - if: $CI_EXTERNAL_PULL_REQUEST_IID
+
+trigger_internal_ci:
+  stage: build
+  image: node
+  tags: 
+    - docker-prod
+  variables:
+    DOWNSTREAM_PROJECT_ID: $DOWNSTREAM_PROJECT_ID
+    DOWNSTREAM_PROJECT_NAME: $DOWNSTREAM_PROJECT_NAME
+  script:
+    - |
+      set -e
+      echo $CI_EXTERNAL_PULL_REQUEST_SOURCE_BRANCH_NAME
+      export TRIGGER_BRANCH=$CI_EXTERNAL_PULL_REQUEST_SOURCE_BRANCH_NAME
+      #Creating a new branch
+        curl --request POST --header \
+        "PRIVATE-TOKEN: $SVC_HLS_RENDER_API" \
+        "${CI_SERVER_URL}/api/v4/projects/$DOWNSTREAM_PROJECT_ID/repository/branches?branch=$CI_EXTERNAL_PULL_REQUEST_SOURCE_BRANCH_NAME&ref=master" || echo "error creating branch $CI_EXTERNAL_PULL_REQUEST_SOURCE_BRANCH_NAME"
+
+      scripts/generate-config.sh
```

Comment:
generate-config.sh script is to dynamically generate a yaml file. gitlab-CI is configured using yaml files. 
Now for our case, we are first creating a new branch in sdk and then triggering the pipeline in sdk on the new branch (aka downstream pipeline). So the CURL command creates the branch but we also need another job to trigger and wait till the job finishes with success or failure.

This triggering could also be be done by REST API, but the work to wait for feedback would have been much complicated and ugly. Therefore we are creating a new YAML config to trigger the downstream pipeline. You can take a look at the HEREDOC section in the script.

Reply:
The reason why we are calling it dynamic is because, the config needs to be created where the branch name of the pipeline is always new (branch name of PR)